prboom2/src/g_game.c

```diff
@@ -1844,7 +1844,7 @@ void G_WorldDone (void)
 
 		  return;
 	  }
-	  else if (gamemapinfo->endpic[0] && (strcmp(gamemapinfo->endpic, "-") != 0))
+	  else if (gamemapinfo->endpic[0] && gamemapinfo->endpic[0] != '-' && !secretexit)
```

Comment:
Why did you have to replace `strcmp()` here? It makes a difference if the `gamemapinfo->endpic` string *is* a minus or merely *starts* with a minus.

Reply:
`'-'` is a special symbol, not user input. This is an implementation detail of PrBoom+'s UMAPINFO.

Reply:
> endpic = "-MYPIC"

In other places we already check only for `[0] == '-'`: https://github.com/coelckers/prboom-plus/blob/784a17d73818cffa4ec3114cef99bcc990d157f4/prboom2/src/e6y.c#L436
https://github.com/coelckers/prboom-plus/blob/784a17d73818cffa4ec3114cef99bcc990d157f4/prboom2/src/g_game.c#L1842

So we need to fix it everywhere?

Reply:
Ouch! No, of course not. Thanks!

Reply:
> Ouch! No, of course not. Thanks!

I thought you'd say we should fix it everywhere :smile: I think we can.

Reply:
> I thought you'd say we should fix it everywhere 😄 I think we can.

Fine, but not in this PR. 😁 (i.e. not too much at once)

We should decide on one way of checking for `'-'` then, and probably even put it into a macro. I am all for the `strcmp()` approach, because it still leaves lumps with names that merely begin with `'-'` as valid choices.